.docker/Dockerfile.cli

```diff
@@ -1,16 +1,33 @@
-FROM govcms8lagoon/govcms8
+##
+# `cli` image.
+#
+# The application is built in this image as it contains build tools like
+# composer and npm.
+#
+# The default PaaS build strategy is:
+#
+#  * wipe the /app directory, from the upstream image build (which is currently SaaS oriented).
+#  * prepare and run composer install inside the `cli` container
+#  * copy in the non-composer files
+#
 
+FROM amazeeio/php:7.2-cli-drupal
+
+# Clean out anything that might be in /app directory and set up the initial files.
 RUN rm -rf /app
-COPY composer.json composer.lock /app/
-COPY scripts /app/scripts
-# Install composer and remove composers cache directory to reduce image size.
-RUN composer install --no-dev && rm -rf ~/.composer/cache
 COPY . /app
```

Comment:
you have to copy app as the last part of the image in order to have it as a last cache layer

Reply:
Why do you need to have it as the last cache layer?

Reply:
Docker layers are rebuilt when files change.

If you have dependencies `COPY` and `composer install` before the rest of the app, the next time some other file is changed, docker does not have to re-run `composer install`. `composer install` is very performance heavy operation, so we want to run only when `compose.json` or `compose.lock` changes. That is why any other files changed will require a layer cache rebuild. Now, if you put `COPY . /app` at the top, your `composer install` will run every time when _any_ file changes. This is a massive overhead and anti-pattern.